components/browser/toolbar/src/test/java/mozilla/components/browser/toolbar/BrowserToolbarTest.kt

```diff
@@ -800,6 +803,62 @@ class BrowserToolbarTest {
         assertEquals(toolbar.displayToolbar.titleView.text, "Mozilla")
     }
 
+    @Test
+    fun `titleView fading is set properly with null attrs`() {
+        val toolbar = BrowserToolbar(testContext)
+        val titleView = toolbar.displayToolbar.titleView
+        val edgeLength = Random.nextInt(1, 24)
+
+        assertFalse(titleView.isHorizontalFadingEdgeEnabled)
+        assertEquals(0, titleView.horizontalFadingEdgeLength)
+
+        titleView.setFadingEdgeLength(edgeLength)
+        titleView.isHorizontalFadingEdgeEnabled = edgeLength > 0
+
+        assertTrue(titleView.isHorizontalFadingEdgeEnabled)
+        assertEquals(edgeLength, titleView.horizontalFadingEdgeLength)
+    }
+
+    @Test
+    fun `titleView fading is set properly with non-null attrs`() {
+        val attributeSet: AttributeSet = Robolectric.buildAttributeSet().build()
+
+        val toolbar = BrowserToolbar(testContext, attributeSet)
+        val titleView = toolbar.displayToolbar.titleView
+        val edgeLength = testContext.resources.getDimensionPixelSize(R.dimen.mozac_browser_toolbar_url_fading_edge_size)
+
+        assertTrue(titleView.isHorizontalFadingEdgeEnabled)
+        assertEquals(edgeLength, titleView.horizontalFadingEdgeLength)
+    }
+
+    @Test
+    fun `urlView fading is set properly with null attrs`() {
+        val toolbar = BrowserToolbar(testContext)
+        val urlView = toolbar.displayToolbar.urlView
+        val edgeLength = Random.nextInt(1, 24)
```

Comment:
I'd prefer constants when testing, since if something breaks its easier to debug when the values are the same. Could you use a for loop with a couple different lengths, such as 1, 12, 24?